pkg/cmd/ssh_azure.go

```diff
@@ -0,0 +1,177 @@
+
+	// add nsg rule
+	a.addNsgRule()
+	fmt.Println("")
+
+	// create public ip
+	a.createPublicIP()
+	fmt.Println("Waiting 5 s until public ip is available.")
+	fmt.Println("")
+	time.Sleep(5 * time.Second)
+
+	// update nic ip-config
+	a.configureNic()
+
+	node := user + "@" + a.PublicIP
+	fmt.Println("Waiting 30 seconds until ports are open.")
+	time.Sleep(30 * time.Second)
+	fmt.Println("(3/4) Establishing SSH connection")
+	fmt.Println("")
+
+	sshCmd := fmt.Sprintf("ssh -i key -o StrictHostKeyChecking=no " + node)
+	cmd := exec.Command("bash", "-c", sshCmd)
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	checkError(err)
+
+	fmt.Println("")
+	fmt.Println("(4/4) Cleanup")
+	a.cleanupAzure()
+}
+
+// fetchAttributes gets all the needed attributes for creating bastion host and its security group with given <nodeIP>.
+func (a *AzureInstanceAttribute) fetchAzureAttributes(nodeIP, path string) {
+	a.NamePublicIP = "sshIP"
+	var err error
+	a.RescourceGroupName, err = ExecCmdReturnOutput("bash", "-c", "cat "+path+" | jq -r '.modules[].outputs.resourceGroupName.value'")
+	checkError(err)
+	a.SecurityGroupName, err = ExecCmdReturnOutput("bash", "-c", "cat "+path+" | jq -r '.modules[].outputs.securityGroupName.value'")
+	checkError(err)
+	targetNode := getNodeForIP(nodeIP)
+	if targetNode == nil {
+		fmt.Println("No node found for ip")
+		os.Exit(2)
+	}
+	// remove invisible controll character which are somehow encoded in the *v1.Node object
+	re := regexp.MustCompile("[[:^ascii:]]")
+	a.NicName = re.ReplaceAllLiteralString(targetNode.Name+"-nic", "")
+
+	// TODO(docktofuture): parse sku type (Basic,Standard,...) from shoot.yaml infrastructure config
+	a.SkuType = "Standard"
+}
+
+// addNsgRule creates a nsg rule to open the ssh port
+func (a *AzureInstanceAttribute) addNsgRule() {
+	fmt.Println("Opened SSH Port.")
+	arguments := fmt.Sprintf("az network nsg rule create --resource-group %s  --nsg-name %s --name ssh --protocol Tcp --priority 1000 --destination-port-range 22", a.RescourceGroupName, a.SecurityGroupName)
+	operate("az", arguments)
+}
+
+// createPublicIP creates the public ip for nic
+func (a *AzureInstanceAttribute) createPublicIP() {
+	var err error
+	fmt.Println("Create public ip")
+	arguments := fmt.Sprintf("az network public-ip create -g %s -n %s --sku %s --allocation-method static", a.RescourceGroupName, a.NamePublicIP, a.SkuType)
+	captured := capture()
+	operate("az", arguments)
+	a.PublicIP, err = captured()
+	checkError(err)
+	fmt.Println(a.PublicIP)
+	data := map[string]interface{}{}
+	dec := json.NewDecoder(strings.NewReader(a.PublicIP))
+	err = dec.Decode(&data)
+	checkError(err)
+	jq := jsonq.NewQuery(data)
+	a.PublicIP, err = jq.String("publicIp", "ipAddress")
+	if err != nil {
+		os.Exit(2)
+	}
+}
+
+// configureNic attaches a public ip to the nic
+func (a *AzureInstanceAttribute) configureNic() {
+	var err error
+	fmt.Println("Add public ip to nic")
```

Comment:
At this step I receive:
```
Add public ip to nic

Different basic sku and standard sku load balancer or public Ip resources in availability set is not allowed
```

Can you reproduce it?

Reply:
Yes, corresponding sku type is parsed out now of lbs.

Reply:
What is the issue? Should we open a separate issue about that with clear steps to reproduce? I guess it is not related to this PR?

Reply:
We need the same sku type for the public ip as for the lb. Therefore we need to parse it for every shoot to be able to create the correct public ip. In the past we only used one sku type and therefore we didn't require that changes. I would like to have this change go with this pr, because we overhauled the complete ssh cmd.